pkg/kubectl/cmd/create.go

```diff
@@ -97,11 +97,11 @@ func RunCreate(f *cmdutil.Factory, out io.Writer, filenames util.StringList) err
 	err = r.Visit(func(info *resource.Info) error {
 		data, err := info.Mapping.Codec.Encode(info.Object)
 		if err != nil {
-			return err
+			return fmt.Errorf("error when creating %q: %v ", info.Source, err)
```

Comment:
Well, it's kinda handy to be able to call kubectl functions from e.g. tests, so let's think carefully before we do that... :)